lib/rules/no-control-regex.js

```diff
@@ -83,8 +96,10 @@ module.exports = {
                 if (regex) {
                     const computedValue = regex.toString();
 
-                    if (hasControlCharacters(computedValue)) {
-                        context.report(node, "Unexpected control character in regular expression.");
+                    const controlCharacters = getControlCharacters(computedValue);
+
+                    if (controlCharacters.length > 0) {
+                        context.report(node, "Unexpected control character(s) in regular expression: " + controlCharacters.join(", ") + ".");
```

Comment:
@ljharb All lint errors have a location (even if you happen not to provide one explicitly), which is the start of the node's first token to the end of its last token. But we can override the location on a per-message basis if we want to flag a particular part of the node. Then formatters and editor integrations can point to the location we provide, which can make the user's life easier. I don't consider it necessary here, but it could be useful in some cases.